### PR TITLE
Recursively remove TL2.fund()

### PR DESCRIPTION
**Pull request by skalermo**

### CODE REVIEW

File: contracts/truefi2/TrueLender2.sol [hunk 1]
```diff
@@ -53,9 +52,6 @@ contract TrueLender2 is ITrueLender2, UpgradeableClaimable {
 
     mapping(ITrueFiPool2 => ILoanToken2[]) public poolLoans;
 
-    // maximum amount of loans lender can handle at once
-    uint256 public maxLoans;
```
Comment (vanruch) [inline contracts/truefi2/TrueLender2.sol:57]: Wouldn't it break storage?

File: test/integration/TrueLender2.test.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ import { mock1Inch_TL2 } from './data'
 
 use(solidity)
 
-describe('TrueLender2', () => {
+xdescribe('TrueLender2', () => {
```
Comment (vanruch) [inline test/integration/TrueLender2.test.ts:30]: Why skipping the test? TL2 will be used until the last loan is repaid so we should keep testing it

Comment (vanruch) [inline test/integration/TrueLender2.test.ts:30]: We need to rewrite this into FTLA tests in following PRs

Comment (skalermo) [reply test/integration/TrueLender2.test.ts:30]: Task filed.

File: contracts/truefi2/TrueLender2.sol [hunk 3]
```diff
@@ -88,27 +88,16 @@ contract TrueLender2 is ITrueLender2, UpgradeableClaimable {
     // minimum prediction market voting period
     uint256 public DEPRECATED__votingPeriod;
 
-    ITrueFiCreditOracle public creditOracle;
+    ITrueFiCreditOracle public DEPRECATED__creditOracle;
```
[the remark this answers is not in the document]
Comment (yuchenlintt) [reply contracts/truefi2/TrueLender2.sol:91]: You can also do this for `DEPRECATED__ratingAgency` in line 66 above.

File: contracts/truefi2/TrueLender2.sol [hunk 4]
```diff
@@ -399,23 +296,6 @@ contract TrueLender2 is ITrueLender2, UpgradeableClaimable {
         return borrowSum;
     }
 
-    /**
```
Comment (yuchenlintt) [inline contracts/truefi2/TrueLender2.sol:402]: The `totalBorrowed()` function in line 381 was only ever called by our new credit model, so we can recursively remove that too. If you prefer, we can do this in a separate PR, although I think it fits here and should be pretty straightforward.

**Review by yuchenlintt — DISMISSED** [review]
I found a couple more state vars and functions to remove, otherwise LGTM!

**Review by vanruch — APPROVED** [review]

**Review by MiksuJak — APPROVED** [review]

**Comment by yuchenlintt** [comment]
LGTM (post-merge).

Confirming I diff checked. The last storage variable on the [mainnet impl](https://etherscan.io/address/0x1bbbf37a055059089d9438566fc5f100e9f08eb6#code) is `votingPeriod`, so it should be safe to delete new storage variables introduced after that point.
